.gitignore

```diff
@@ -60,3 +60,10 @@ book/
 # complement
 /complement-*
 /master.tar.gz
+
+# rust
+/target/
+/synapse/*.so
```

Comment:
do we care about ... dylibs? dlls? Whatever those other OS people do :)

Reply:
Yeah, probably I guess? Though do we support other OSes?

Reply:
Yes, I'm on macOS (and others who occasionally contribute to Synapse use macOS too) -- are you mostly just hoping I run this and make sure it doesn't break / what other files we might need to ignore?

Reply:
Yeah, pretty much

Reply:
On macOS this did seem to only generate a `.so` file, but it was under the `build/` directory. This was when installing an editable install via `pip`, so might be OK if you use poetry?

Reply:
`/build/` is ignored separately fwiw, so thats fine.